test/monitoring/aws-ec2/EC2Monitoring.test.ts

```diff
@@ -2,70 +2,90 @@ import { Stack } from "aws-cdk-lib";
 import { Template } from "aws-cdk-lib/assertions";
 import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
 
-import { EC2Monitoring } from "../../../lib";
+import { EC2Monitoring, InstanceFamilyTypes } from "../../../lib";
 import { addMonitoringDashboardsToStack } from "../../utils/SnapshotUtil";
 import { TestMonitoringScope } from "../TestMonitoringScope";
 
-test("snapshot test: all instances, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter + ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
+function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
+  return Object.keys(obj).filter((k) => Number.isNaN(+k)) as K[];
+}
+
+test.each(enumKeys(InstanceFamilyTypes))(
+  "snapshot test: all instances, no alarms. Instance family type: %p",
+  (instanceFamilyType) => {
+    const stack = new Stack();
+
+    const scope = new TestMonitoringScope(stack, "Scope");
+
+    const monitoring = new EC2Monitoring(scope, {
+      alarmFriendlyName: "EC2",
+      instanceFamilyType: InstanceFamilyTypes[instanceFamilyType],
```

Comment:
It'd be good to capture default behaviour in a separate test too.

Reply:
That's a good suggestion, I'll do that.